lib/commands/create.js

```diff
@@ -34,6 +34,10 @@ var command = {
       return done(new ConfigurationError("Please specify the name of item to create. Example: truffle create contract MyContract"));
     }
 
+    if (!/^[a-zA-Z_$][a-zA-Z_$0-9]*$/.test(name)) {
+      return done(new ConfigurationError("The name " + name + " is invalid. Please enter a valid name to create."));
+    }
```

Comment:
Looks good! If possible could you modify the error message to describe a valid string. Maybe `Please enter an alpha-numeric string.` instead of `Please enter a valid name ... `?

Reply:
For future reference or further discussion - stack exchange debate on the relevant regex [here](https://stackoverflow.com/questions/2008279/validate-a-javascript-function-name). I'm happy with the one proposed fwiw - we're just trying to help people create a file.

Reply:
Updated the error!

I chose the regex from the Solidity language definition. Even if Javascript changed and `my-contract` became a valid function name, it still wouldn't be valid Solidity.

Reply:
Oh perfect, makes sense. Thanks for linking to that.